docs/source/reference/kubernetes/kubernetes-setup.rst

```diff
@@ -215,6 +215,99 @@ Once the cluster is deployed and you have placed your kubeconfig at ``~/.kube/co
+clusters running on Kubernetes, we support either of two modes to expose ports:
+
+* :ref:`LoadBalancer Service <kubernetes-loadbalancer>` (default)
+* :ref:`Nginx Ingress <kubernetes-ingress>`
+
+
+By default, SkyPilot creates a `LoadBalancer Service <https://kubernetes.io/docs/concepts/services-networking/service/>`__ on your Kubernetes cluster to expose the port.
+
+If your cluster does not support LoadBalancer services, SkyPilot can also use `an existing Nginx IngressController <https://kubernetes.github.io/ingress-nginx/>`_ to create an `Ingress <https://kubernetes.io/docs/concepts/services-networking/ingress/>`_ to expose your service.
+
+.. _kubernetes-loadbalancer:
+
+LoadBalancer Service
+^^^^^^^^^^^^^^^^^^^^
+
+This mode exposes ports through a Kubernetes `LoadBalancer Service <https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer>`__. This is the default mode used by SkyPilot.
+
+
+To use this mode, you must have a Kubernetes cluster that supports LoadBalancer Services:
+
+* On Google GKE, Amazon EKS or other cloud-hosted Kubernetes services, this mode is supported out of the box and no additional configuration is needed.
+* On bare metal and self-managed Kubernetes clusters, `MetalLB <https://metallb.universe.tf/>`_ can be used to support LoadBalancer Services.
+
+When using this mode, SkyPilot will create a single LoadBalancer Service for all ports that you expose on a cluster.
+Each port can be accessed using the LoadBalancer's external IP address and the port number. Use :code:`sky status --endpoints <cluster>` to view the external endpoints for all ports.
+
+.. note::
+    In cloud based Kubernetes clusters, this will automatically create an external Load Balancer.     GKE creates a (`pass-through load balancer <https://cloud.google.com/kubernetes-engine/docs/concepts/service-load-balancer>`__)
+    and AWS creates a `Network Load Balancer <https://docs.aws.amazon.com/eks/latest/userguide/network-load-balancing.html>`__). These load balancers will be automatically terminated when the cluster is deleted.
+
+.. note::
+    The default LoadBalancer implementation in EKS selects a random port from the list of opened ports for the
+    `LoadBalancer's health check <https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html>`_. This can cause issues if the selected port does not have a service running behind it.
+
+
+    For example, if a SkyPilot task exposes 5 ports but only 2 of them have services running behind them, EKS may select a port that does not have a service running behind it and the LoadBalancer will not pass the healthcheck. As a result, the service will not be assigned an external IP address.
+
+    To work around this issue, make sure all your ports have services running behind them.
+
+.. note::
+    LoadBalancer services are not supported on kind clusters created using :code:`sky local up`.
```

Comment:
Just a quick thought, maybe we can do it in the future, but is it possible that we can have `sky local up` with the nginx ingress setup by default, so the user can directly try the open ports on the cluster that is `local up`'ed

Reply:
Good idea! tracking here - https://github.com/skypilot-org/skypilot/issues/3112 : )